### ISSUE TITLE
REST endpoint(s) for querying builds based on branches

### ISSUE THREAD

**Issue by donny-dont** [drone/drone#1495]
It would be great if there was a way to query for builds based on a branch. In swagger this would look like

```
/repos/{owner}/{name}/{branch}/builds:
    get:
      parameters:
        - name: owner
          in: path
          type: string
          description: owner of the repository
        - name: name
          in: path
          type: string
          description: name of the repository
        - name: branch
          in: path
          type: string
          description: the branch of the buil
      tags:
        - Builds
      summary: Get recent builds
      description: Returns recent builds for the repository based on name.
      security:
        - accessToken: []
      responses:
        200:
          description: The recent builds on the branch.
          schema:
            type: array
            items:
              $ref: "#/definitions/Build"
        404:
          description: |
            Unable to find the Repository in the database
```

Additionally it may be useful to get a listing of all the branches that have been seen or maybe the latest build on a per branch basis. This would also potentially have the pagination ala #1494.

**Comment by tboerger** [comment]
Sounds fine

**Comment by strk** [comment]
There was a previous proposal for a more general `filter` support here: https://github.com/drone/drone-ui/issues/63#issuecomment-488752422

What happened about that ?

**Comment by bradrydzewski** [comment]
added query parameters to the builds endpoint to list builds for a specific branch. This is available in master and will be included in the 1.5 release.

```
GET /api/repos/{namespace}/{name}/builds?branch={branch}
```
